### DISCUSSION TITLE
Admin auth

### DISCUSSION THREAD

**Discussion by ZelimDamian** [supabase/supabase#14864] [Questions]
Hi. Had a few questions about auth:

1) How do I reset user password without sending emails and/or OTP if they forgot their password? 

2) How to force login as another user? Example (https://blitzjs.com/docs/impersonation)

3) As with the above. I need to generate a session for a user when implementing custom auth through edge functions. I am aware that you can mint your own access token, but `auth.setSession` requires both access and refresh tokens. The refresh token has to be placed into the DB but I cann't find a way to do that except raw SQL insert which is undesirable cause this relies on the underlying schema staying constant (don't know for sure if that will even work).

**Comment by ioucyf** [comment]
I have the same problem. I don't know why it's not possible for admin to issue new sessions for users; like every other method without the need for us to mint our own jwts. 

That would've solved everything for so many people.

**Comment by bombillazo** [comment]
At the moment, those need to be managed on your own with custom SQL to Edge Function logic, unfortunately. 

Yeah, GoTrue is an excellent OS starting point for auth, but it still needs more robust and advanced customization and management options. :/ In addition to those you specified, we've wanted to create unclaimed users that can later be claimed with certain confirmation/verification processes using the same auth API.
